pkg/controller/controller_test.go

```diff
@@ -63,83 +59,139 @@ var _ = Describe("Controller", func() {
-				Expect(pod).ToNot(BeNil(), fmt.Sprintf("importer pod %q missing", exptPod))
-				Expect(pod.GenerateName).To(HavePrefix(exptPod))
+
+		tests := []testT{
+			{
+				descr:       "pvc, endpoint, blank ns: controller creates importer pod",
+				ns:          "", // blank used originally in these unit tests
+				name:        "test-pvc",
+				qops:        opAdd,
+				annotations: map[string]string{AnnEndpoint: "http://www.google.com"},
+				expectError: false,
+			},
+			{
+				descr:       "pvc, endpoint, non-blank ns: controller creates importer pod",
+				ns:          "ns-a",
+				name:        "test-pvc",
+				qops:        opAdd,
+				annotations: map[string]string{AnnEndpoint: "http://www.google.com"},
+				expectError: false,
+			},
+			{
+				descr:       "pvc, blank endpoint: controller does not create importer pod",
+				ns:          "",
+				name:        "test-pvc",
+				qops:        opAdd,
+				annotations: map[string]string{},
+				expectError: true,
+			},
+			{
+				descr:       "updated pvc should process",
+				ns:          "ns-a",
+				name:        "test-pvc",
+				qops:        opUpdate,
+				annotations: map[string]string{AnnEndpoint: "http://www.google.com"},
+				expectError: false,
+			},
+			{
+				descr:       "updated pvc should not process based on annotation AnnImportPod indicating already been processed",
+				ns:          "ns-a",
+				name:        "test-pvc",
+				qops:        opUpdate,
+				annotations: map[string]string{AnnEndpoint: "http://www.google.com", AnnImportPod: "importer-test-pvc"},
+				expectError: true,
+			},
+		}
+
+		for _, test := range tests {
+			ns := test.ns
+			pvcName := test.name
+			ops := test.qops
+			annotations := test.annotations
+			fullname := pvcName
+			if len(ns) > 0 {
+				fullname = fmt.Sprintf("%s/%s", ns, pvcName)
 			}
+			exptPod := fmt.Sprintf("importer-%s-", pvcName)
+			exptErr := test.expectError
+
+			It(test.descr, func() {
+				By(fmt.Sprintf("creating in-mem pvc %q with endpt anno=%q", fullname, annotations))
+				pvc := createInMemPVC(ns, pvcName, annotations)
+				By("Creating the controller")
+				setUpInformers(pvc, nil, ns, ops, stop)
+				controller.ProcessNextPvcItem()
+				By("checking if importer pod is present")
+				pod, err := getImporterPod(fakeClient, ns, exptPod)
+				if exptErr {
+					By("Expecting no pod created")
+					Expect(err).ToNot(BeNil(), fmt.Sprintf("importer pod %s... should not exist\n", exptPod))
+				} else {
+					By("Expecting pod created")
+					Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("importer pod: %v\n", err))
+					Expect(pod).ToNot(BeNil(), fmt.Sprintf("importer pod %q missing", exptPod))
+					Expect(pod.GenerateName).To(HavePrefix(exptPod))
+				}
+			})
+		}
+	})
+
+	Context("when annotating pod status in pvc", func() {
+
+		var (
+			pvc *v1.PersistentVolumeClaim
+			//pod     *v1.Pod
+			pvcName string
+			ns      string
+			ann     map[string]string
+		)
+
+		BeforeEach(func() {
+			pvcName = "test-pvc"
```

Comment:
maybe this can be a constant?

Reply:
down the road yes, all the code is scratch at the moment just to debug the Update() problem